libsolidity/ast/Types.cpp

```diff
@@ -2224,6 +2224,15 @@ u256 StructType::storageSize() const
 	return max<u256>(1, members(nullptr).storageSize());
 }
 
+bool StructType::canLiveOutsideStorage() const
+{
+	bool result = true;
+	for (auto const& member: m_struct.members())
+		if (!recursive())
```

Comment:
This allows code like the following:

```solidity
contract C {
	struct S {
		mapping (uint => S) map;
	}

	mapping (uint => S) structs;

	function f() public view {
		S memory test = structs[1];
		test;
	}
}
```

Reply:
I will remove the if condition. Can't think of a reason it would cause problems.

Reply:
```solidity
contract C {
    struct S { uint a; S[2][] sub;}
    function f() public pure returns (uint) {
		S memory test;
		test;
    }
}
```
This is a code that causes problems.